imap_processing/mag/l0/decom_mag.py

```diff
@@ -53,9 +62,89 @@ def decom_packets(packet_file_path: str) -> list[MagL0]:
                     ]
                     data_list.append(MagL0(CcsdsData(packet.header), *values))
         except ReadError as e:
-            logger.error(
+            logger.warning(
                 f"Found error: {e}\n This may mean reaching the end of an "
                 f"incomplete packet."
             )
 
         return data_list
+
+
+def export_to_xarray(l0_data: list[MagL0]):
+    """Generate xarray files for "raw" MAG CDF files from MagL0 data.
+
+    Mag outputs "RAW" CDF files just after decomming. These have the immediately
+    post-decom data, with raw binary data for the vectors instead of vector values.
+
+    Parameters
+    ----------
+    l0_data: list[MagL0]
+        A list of MagL0 datapoints
+    """
+    # TODO split by mago and magi using primary sensor
+    # TODO split by norm and burst
+    norm_data = defaultdict(
+        list
+    )  # dict.fromkeys(dataclasses.asdict(l1a_data[0]).keys(), [])
+    burst_data = norm_data.copy()
+
+    for datapoint in l0_data:
+        if datapoint.ccsds_header.PKT_APID == Mode.NORM:
+            for key, value in dataclasses.asdict(datapoint).items():
+                if key != "ccsds_header":
+                    norm_data[key].append(value)
+        if datapoint.ccsds_header.PKT_APID == Mode.BURST:
+            burst_data["SHCOARSE"].append(datapoint.SHCOARSE)
+            burst_data["raw_vectors"].append(datapoint.VECTORS)
+
+    # Used in L1A vectors
+    direction = xr.DataArray(
+        np.arange(194),
```

Comment:
Is this a placeholder for now to be filled in with real data later? Or is this a coordinate that is "fixed" in a sense? I was expecting to see something related to `norm_data["raw_vectors"]`, so it might help to put a quick comment in one way or the other.

Reply:
That is a good question that I don't fully remember the answer too. I think this is a variable coordinate that corresponds to the length of the vector (for each time stamp, there should be a vector of length 4 - x, y, z, and sensor range) but for these L0 RAW CDF files, there is instead a long string of bytes. So I think this was mostly me testing that I could get that 2d array in the CDF, but this format isn't correct for this output.